backend/src/akvo/lumen/lib/aggregation/bubble.clj

```diff
@@ -29,36 +31,26 @@
-        column-label-name (get column-label "columnName")
-        column-label-title (get column-label "title")
-        column-bucket (utils/find-column columns (get query "bucketColumn"))
-        column-bucket-type (get column-bucket "type")
-        column-bucket-name (get column-bucket "columnName")
-        column-bucket-title (get column-bucket "title")
+        column-size (keywordize-keys (utils/find-column columns (get query "metricColumn")))
+        column-bucket (keywordize-keys (utils/find-column columns (get query "bucketColumn")))
         max-points 2500
-        have-aggregation (boolean column-bucket)
-        aggregation-method (if column-size
-                             (get query "metricAggregation")
-                             "count")
-
-        sql-text-with-aggregation (str "SELECT "
-                                       (sql-aggregation-subquery aggregation-method "%1$s" (or column-size-type column-bucket-type))
-                                       " AS size, "
-                                       "%7$s AS label FROM (SELECT * FROM %2$s WHERE %3$s ORDER BY random() LIMIT %5$s)z GROUP BY %7$s")
-        sql-text-without-aggregation "
-        SELECT * FROM (SELECT * FROM (SELECT %1$s AS size, %6$s AS label FROM %2$s WHERE %3$s)z ORDER BY random() LIMIT %5$s)zz ORDER BY zz.x"
-        sql-text (if have-aggregation sql-text-with-aggregation sql-text-without-aggregation)
-      sql-response (run-query tenant-conn table-name sql-text (or column-size-name (get column-bucket "columnName")) filter-sql aggregation-method max-points column-label-name column-bucket-name)]
+        aggregation-method (if column-size (get query "metricAggregation") "count")
+        sql-text (str "SELECT "
+                      (sql-aggregation-subquery aggregation-method "%1$s" (or (:type column-size) (:type column-bucket)))
+                      " AS size, "
+                      "%6$s AS label FROM (SELECT * FROM %2$s WHERE %3$s ORDER BY random() LIMIT %5$s)z GROUP BY %6$s")
+        sql-response (run-query tenant-conn table-name
+                                sql-text
+                                (or (:columnName column-size) (:columnName column-bucket))
+                                filter-sql
+                                aggregation-method
+                                max-points
+                                (:columnName column-bucket))]
     (lib/ok
-     {"series" [{"key" column-size-title
-                 "label" column-size-title
+     {"series" [{"key" (:title column-size)
+                 "label" (:title column-size)
                  "data" (mapv (fn [[size-value label]] {"value" size-value}) sql-response)
-                 "metadata"  {"type" column-size-type}}]
-      "common" {"metadata" {"type" column-label-type "sampled" (= (count sql-response) max-points)}
+                 "metadata"  {"type" (:type column-size)}}]
+      "common" {"metadata" {"type" nil
```

Comment:
@finnfiddle which type should we return here? as far as we are not using  column-label anymore in this visualisation ...

Reply:
@finnfiddle ping!